Services/UI/OJS.Services.Ui.Data/Implementations/ContestsDataService.cs

```diff
@@ -43,7 +43,8 @@ public async Task<IEnumerable<TServiceModel>> GetAllPast<TServiceModel>()
 
         public Task<Contest?> GetByIdWithProblems(int id)
             => this.DbSet
-                .Include(c => c.ProblemGroups.Select(pg => pg.Problems))
+                .Include(c => c.ProblemGroups)
+                .ThenInclude(pg => pg.Problems)
                 .FirstOrDefaultAsync(c => c.Id == id);
```

Comment:
Why not AutoMapper?

Reply:
The whole results logic is copied from old judge.